pkg/controller/versions/strategy_v1_0.go

```diff
@@ -79,12 +91,24 @@ func (v *versionStrategyV1_0) Validate(ctx context.Context, cl client.Client, sm
 	return utilerrors.NewAggregate(allErrors)
 }
 
-func (v *versionStrategyV1_0) ValidateDowngrade(ctx context.Context, cl client.Client, smcp *v1.ServiceMeshControlPlane) error {
+func (v *versionStrategyV1_0) ValidateV2(ctx context.Context, cl client.Client, smcp *v2.ServiceMeshControlPlane) error {
```

Comment:
I'm starting to think that maybe the v2 resource should not be used to deploy v1 control planes. Not sure how to enforce that though, with Kubernetes only accepting one resource version to be the default at all times

Reply:
It's something to consider, although it would be ideal if we could ensure as many settings as possible are mapped.  Part of the reason for having v2 is to declare a known set of supported options.  It's certainly possible that some fringe settings could be missed in the new schema.